File: lib/prompts.ts
import type { RetrievedChunk } from "./retrieval";

export function buildRagPrompt(question: string, chunks: RetrievedChunk[]) {
  const context = chunks
    .map((chunk, index) => {
      const source = chunk.metadata?.source ?? "pasted text";
      const chunkIndex = chunk.metadata?.chunkIndex ?? chunk.id;

      return `Source ${index + 1} (${source}, chunk ${chunkIndex}, similarity ${chunk.similarity.toFixed(
        3,
      )}):\n${chunk.content}`;
    })
    .join("\n\n---\n\n");

  return `You are a careful document question-answering assistant.

Use only the provided context to answer the user's question.
If the context does not contain enough information, say that the document context does not provide enough information.
Be concise, specific, and cite the source numbers you used.

Context:
${context}

Question:
${question}

Answer:`;
}

File: backend/app/prompts.py
from app.schemas import RetrievedChunk


def build_rag_prompt(question: str, chunks: list[RetrievedChunk]) -> str:
    context_parts: list[str] = []

    for index, chunk in enumerate(chunks):
        metadata = chunk.metadata or {}
        source = metadata.get("source", "pasted text")
        chunk_index = metadata.get("chunkIndex", chunk.id)
        context_parts.append(
            f"Source {index + 1} ({source}, chunk {chunk_index}, "
            f"similarity {chunk.similarity:.3f}):\n{chunk.content}"
        )

    context = "\n\n---\n\n".join(context_parts)

    return f"""You are a careful document question-answering assistant.

Use only the provided context to answer the user's question.
If the context does not contain enough information, say that the document context does not provide enough information.
Be concise, specific, and cite the source numbers you used.

Context:
{context}

Question:
{question}

Answer:"""
